src/cargo/util/config.rs

```diff
@@ -573,8 +674,8 @@ impl Config {
 
     /// Loads credentials config from the credentials file into the ConfigValue object, if present.
     fn load_credentials(&self, cfg: &mut ConfigValue) -> CargoResult<()> {
-        let home_path = self.home_path.clone().into_path_unlocked();
-        let credentials = home_path.join("credentials");
+        let config_path = self.dirs.config_dir.clone().into_path_unlocked();
+        let credentials = config_path.join("credentials");
```

Comment:
Hm, and what is the "correct" place for credentials? `config` might be not the right place, because people sometimes publish it... I suggest raising this question on the RFC thread, if it wasn't raised already.